Tests/MetricsTests/TestMetrics.swift

```diff
@@ -35,8 +35,12 @@ internal final class TestMetrics: MetricsFactory {
         return self.make(label: label, dimensions: dimensions, registry: &self.recorders, maker: maker)
     }
 
-    public func makeTimer(label: String, dimensions: [(String, String)]) -> TimerHandler {
-        return self.make(label: label, dimensions: dimensions, registry: &self.timers, maker: TestTimer.init)
+    public func makeTimer(label: String, storageUnit: TimeUnit, dimensions: [(String, String)]) -> TimerHandler {
+        self.lock.withLock {
```

Comment:
define a `maker` func like in L32

Reply:
I moved away from the maker function here since it does not allow for passing through of the `displayUnit`.

Reply:
I got that, yes. The problem here is that the maker function has to have the signature of `(String, [String, String]) -> Item` so I can not pass through the `preferredDisplayUnit`

Reply:
Ah, yes. My bad. I'll update it!